### ISSUE TITLE
Getting Bad request when there are multiple recipients

### ISSUE THREAD

**Issue by sharath004** [sendgrid/sendgrid-csharp#896]
#### Issue Summary

Sendgrid API throws a bad request upon execution.
 

![badrequest](https://user-images.githubusercontent.com/8668771/59998382-99294d80-96a2-11e9-9eec-fa7e81c07b64.png)

#### Steps to Reproduce
1. I have around 600 bcc receipients.
2. I have included both subject and TO(They do have values)
3. I pass in the key(guid) to the proxy but I get a bad request on executing it
4. The key that I am using has full access

#### Technical details:

Execute(message).ConfigureAwait(false);

 static async Task Execute(MailMessage message, ISMTPLog _smtpLog, FusionBLL.Prospector.Email email, Guid publicGuid, string _mailServer)
        {
            var apiKey = ConfigurationManager.AppSettings["SendGridKey"];
            var client = new SendGridClient(apiKey);

            var msg = new SendGridMail.SendGridMessage()
            {
                From = new SendGridMail.EmailAddress(message.From.Address, message.From.DisplayName),
                Subject = message.Subject,
                HtmlContent = message.Body
            };
            foreach (var to in message.To)
            {
                msg.AddTo(new SendGridMail.EmailAddress(to.Address, to.DisplayName));
            }
            foreach (var cc in message.CC)
            {
                msg.AddCc(new SendGridMail.EmailAddress(cc.Address, cc.DisplayName));
            }
            foreach (var bcc in message.Bcc)
            {
                msg.AddBcc(new SendGridMail.EmailAddress(bcc.Address, bcc.DisplayName));
            }
            foreach (var attachment in message.Attachments)
            {
                byte[] bytes;
                using (var memoryStream = new MemoryStream())
                {
                    attachment.ContentStream.CopyTo(memoryStream);
                    bytes = memoryStream.ToArray();
                }
                string base64 = Convert.ToBase64String(bytes);
                msg.AddAttachment(attachment.Name, base64, attachment.ContentType.MediaType, null, attachment.ContentId);
            }
            var response = await client.SendEmailAsync(msg);
          
        }

**Comment by Fieora** [comment]
Seems like it could be relevant here, but you are responsible for making sure there are no duplicate addresses at least between the to and the cc fields, I'm unsure as of writing this if bcc is the same.

To make it easier on yourself, if message.To is a `List<EmailAddress>` (enumerable not an array) you can call the plural function msg.AddTos (Ccs, Bccs) instead of freaching them.

tl;dr If the email addresses aren't unique the await client.SendEmailAsync(msg) will return BadRequest.

Good Luck!

**Comment by aevitas** [comment]
Once #905 gets merged, you'll be able to call `recipients.Distinct()` to ensure there are no duplicates in your recipient list.

**Comment by thinkingserious** [comment]
Thanks for jumping in and helping @Fieora & @aevitas!

And thanks to @aevitas for the PR!

**Comment by thinkingserious** [comment]
Since there has been no activity on this issue since March 1, 2020, we are closing this issue. Please feel free to reopen or create a new issue if you still require assistance. Thank you!
